source/jormungandr/jormungandr/equipments/equipment_provider_manager.py

```diff
@@ -122,32 +124,49 @@ def update_config(self):
                 # If the provider added in db is also defined in legacy, delete it.
                 self._equipment_providers_legacy.pop(provider.id, None)
 
-    def manage_equipments(self, response):
+    def manage_equipments(self, response, api):
         """
         Call equipment details provider to update response
         :param response: the pb response received
+        :param api: the API type
         :return: response: the pb response updated with equipment details
         """
+        if api == type_pb2.API.Value('equipment_reports'):
+            stop_area_equipments = [sae for er in response.equipment_reports for sae in er.stop_area_equipments]
```

Comment:
for consistency with `get_from_to_stop_points_of_journeys` (and performances)  you could use an operator here.

```suggestion
            stop_area_equipments = (sae for er in response.equipment_reports for sae in er.stop_area_equipments)
```

Reply:
:+1: